### ISSUE TITLE
Crash on OSX on v03

### ISSUE THREAD

**Issue by lukas1** [featurecat/lizzie#90]
Setup:
OSX 10.13.4 (17E199)
JDK 10

Steps to reproduce:

1. Download lizzie release v0.3
2. Compile featurecat/leela-zero
3. Copy and paste leelaz to lizzie directory
4. Run lizzie from command line `java -jar Lizzie.jar`
5. Play a few moves
6. Press `Space` in hopes of turning pondering. Observation: The heatmap is not showing. But from the output in terminal, it looks like leela zero did return a few candidate moves:

```
Wavefront/Warp size: 8
Max workgroup size: 512
Max workgroup dimensions: 512 512 512 
= 


 Q17 ->       1 (V: 48.32%) (N: 10.01%) PV: Q17 
 C16 ->       0 (V:  0.00%) (N:  9.55%) PV: C16 
1.5 average depth, 2 max depth
1 non leaf nodes, 1.00 average children

2 visits, 723 nodes

= 

= 


 D16 ->       1 (V: 53.38%) (N: 21.34%) PV: D16 
 D17 ->       0 (V:  0.00%) (N: 15.25%) PV: D17 
1.5 average depth, 2 max depth
1 non leaf nodes, 1.00 average children

2 visits, 721 nodes

= 

= 


 C17 ->       1 (V: 47.49%) (N: 30.41%) PV: C17 
 R16 ->       0 (V:  0.00%) (N: 15.13%) PV: R16 
1.5 average depth, 2 max depth
1 non leaf nodes, 1.00 average children

2 visits, 719 nodes

= 

= 


  D4 ->       1 (V: 53.43%) (N: 50.34%) PV: D4 
  C4 ->       0 (V:  0.00%) (N: 22.76%) PV: C4 
1.5 average depth, 2 max depth
1 non leaf nodes, 1.00 average children

2 visits, 717 nodes

= 

= 


  C3 ->       1 (V: 43.94%) (N: 48.58%) PV: C3 
 C17 ->       0 (V:  0.00%) (N: 36.73%) PV: C17 
1.5 average depth, 2 max depth
1 non leaf nodes, 1.00 average children

2 visits, 715 nodes

= Leela Zero
```

7. Shut down Lizzie (either by command line or in UI)
8. Start lizzie again. Crashes right at the start with:
```
java.lang.NullPointerException
	at wagner.stephanie.lizzie.analysis.Leelaz.parseLine(Leelaz.java:122)
	at wagner.stephanie.lizzie.analysis.Leelaz.read(Leelaz.java:155)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

Full logs of Lizzie startup when it crashes:

```
time_left b 0 0
Using 2 thread(s).
RNG seed: 6708713849056509036
Using per-move time margin of 0.00s.
Detecting residual layers...v1...128 channels...10 blocks.
Initializing OpenCL.
Detected 1 OpenCL platforms.
Platform version: OpenCL 1.2 (Mar 15 2018 21:59:37)
Platform profile: FULL_PROFILE
Platform name:    Apple
Platform vendor:  Apple
Device ID:     0
Device name:   Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
Device type:   CPU
Device vendor: Intel
Device driver: 1.1
Device speed:  2500 MHz
Device cores:  4 CU
Device score:  512
Device ID:     1
Device name:   HD Graphics 4000
Device type:   GPU
Device vendor: Intel
Device driver: 1.2(Mar 15 2018 22:04:25)
Device speed:  1100 MHz
Device cores:  16 CU
Device score:  612
Selected platform: Apple
Selected device: HD Graphics 4000
with OpenCL 1.2 capability.
Loaded existing SGEMM tuning.
Wavefront/Warp size: 8
Max workgroup size: 512
Max workgroup dimensions: 512 512 512 
= 

java.lang.NullPointerException
	at wagner.stephanie.lizzie.analysis.Leelaz.parseLine(Leelaz.java:122)
	at wagner.stephanie.lizzie.analysis.Leelaz.read(Leelaz.java:155)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

**Comment by featurecat** [comment]
Thank you. I found the bug. it looks like we fixed this since then, could you recompile the jar from this repository? otherwise i can release the jar that you can test to see if the problem is fixed

**Comment by lukas1** [comment]
I've tried to recompile the current repository version and you're right, the problem has been fixed. So I believe this issue can be closed. Now, I'm not going to do that just yet and I'll let you decide what to do with this issue, because I found another issue.

Heatmap of pondered moves does not show on first launch of lizzie (first, meaning, there's no `leelaz_opencl_tuning` or `lizzie.properties` file in the jar directory). Subsequent launches seem to work just fine.

I can open another issue if you prefer that and we could close this one. That seems like a logical step to me, but I don't want to spam you :-) The issue itself is not a big deal honestly, especially if one knows that subsequent launches work fine.

Also I have one question, how do you actually generate that `network` file? I was not able to find that one out. I had to copy `network` file from the v03 release zip. I've looked at leela-zero repo as well, but nothing obvious jumped at me. 

Thanks and I apologize for spamming so much.

**Comment by featurecat** [comment]
Oh! I didn't realize network wasn't clear... you can get a new one from http://zero.sjeng.org/best-network. You can view all networks at http://zero.sjeng.org. The specific network used for Lizzie 0.3 was LZ109. The current network is LZ123, so I'd highly recommend updating the network file. I think I might make Lizzie update the network automatically in the near future.

> Heatmap of pondered moves does not show on first launch of lizzie (first, meaning, there's no leelaz_opencl_tuning or lizzie.properties file in the jar directory). Subsequent launches seem to work just fine.

I think what's happening here is Leela Zero is generating the leelaz_opencl_tuning file. I deleted both lizzie.properties and leelaz_opencl_tuning, and when I started the program, it took 15 seconds before the heatmap appeared. I should probably add a "loading" icon.

**Comment by lukas1** [comment]
Oh yes, you're right, it will start the heat map even on the first run, except that on my machine (6 years old Macbook Pro) it runs really slowly, so the `leelaz_opencl_tuning` is generating a bit longer than on your machine. 

Thanks for clarifying on the network file, the bit that confused me is, that in the v03 zip it is marked as executable file, so I thought it was somehow post-processed. I didn't know it's the same weights file that I could download from http://zero.sjeng.org/

**Comment by dfannius** [comment]
> I can open another issue if you prefer that and we could close this one.

Issues are not a scarce resource :) In basically any project it is best practice to have an issue for each subject. Then you can write things like "Fixes issue #90" and close it, rather than "Fixes the second thing mentioned in issue #90" and then have to pore through the issue looking to see what parts of it are active or inactive. (Of course it is no big deal, just letting you know not to feel bad about "spamming" in the future here or anywhere.)

**Comment by lukas1** [comment]
@dfannius  sure, I just wasn't too sure, this being my first opening of an issue on this github project in particular.

anyhow, I do believe, that this issue can actually be closed, as the crash is clearly resolved on the newest version of lizzie and if necessary I can open an issue about missing progress indicator on first run of lizzie, but I can imagine that this task has very low priority, given it's an issue only on the first run.
